apps/callscreen/js/call_screen.js

```diff
@@ -379,25 +377,32 @@ var CallScreen = {
   },
 
   showClock: function cs_showClock(now) {
-    LazyL10n.get(function localized(_) {
+    var self = this;
+    var timeFormatId = navigator.mozHour12 ? 'shortTimeFormat12'
+                                           : 'shortTimeFormat24';
+
+    return Promise.all([
+      navigator.mozL10n.formatValue(timeFormatId),
+      navigator.mozL10n.formatValue('longDateFormat')
+    ]).then(function(results) {
       var f = new navigator.mozL10n.DateTimeFormat();
-      var timeFormat = window.navigator.mozHour12 ? _('shortTimeFormat12') :
-                                                    _('shortTimeFormat24');
+      var timeFormat = results[0].replace('%p', '<span>%p</span>');
+      var dateFormat = results[1];
+
       /* In 12h format we don't want to display the AM/PM component on the
        * lockscreen. However there's no way to obtain such a string using
        * DateTimeFormat(). What we do instead is to inject a fake <span> tag
        * around the AM/PM component which we later remove from the string using
        * a regular expression. */
       timeFormat = timeFormat.replace('%p', '<span>%p</span>');
-      var dateFormat = _('longDateFormat');
 
       var timeText =
         f.localeFormat(now, timeFormat).replace(/\s<span>.*<\/span>/, '');
       var dateText = f.localeFormat(now, dateFormat);
 
-      this.lockedClockTime.textContent = timeText;
-      this.lockedDate.textContent = dateText;
-    }.bind(this));
+      self.lockedClockTime.textContent = timeText;
+      self.lockedDate.textContent = dateText;
+    });
```

Comment:
showClock has been updated in my lockscreen Intl patch so this change is not needed